### DISCUSSION TITLE
Resize GiF

### DISCUSSION THREAD

**Discussion by Yogesh11** [SDWebImage/SDWebImage#3541] [Q&A] [answered]
Hi, we are getting gif as an image URL from the API. We have to set the same gif image in tabItem..As you know tabItem will take only UIImage object so do you support resize gif ? we are using SDWebImage (5.9.2):

**Comment by dreampiggy** [comment, marked as answer]
You can either use thumbnail decoding (which means, the full size image is even not created, only the thubmnail one is created in memory), the cache key is different from the full one.

or transform (the `resizeTransformer`, which means, the full size image is firstly created then resize down, two image object in memory)

See wiki carefully: https://github.com/SDWebImage/SDWebImage/wiki/Advanced-Usage#thumbnail-decoding-550

**Reply by Yogesh11** [reply]
let thumbnailData = SDImageIOCoder.shared.encodedData(with: image, format: .GIF, options: [.encodeMaxPixelSize: CGSize(width: 24, height: 24)])


Since i used this for gif.. its giving me a static image... Gif is not playing.

**Reply by dreampiggy** [reply]
Why not use `encodedData(with: frames, format:, loopCount:, options:)` ?

**Reply by dreampiggy** [reply]
If the image class is `SDAnimatedImage`, you shouldn't use re-encode. The image data buffer is available from `animatedImageData` property.

If you want to re-scale a `SDAnimatedImage` (I don't think people do this, but still possible). You need for-in loop to grap each frame and call that `encodedData(with:frames)` animated image encode API. See more in PR description of 5.15.0

Or I can update this into the wiki

**Comment by Yogesh11** [comment]
Hi @Dream Piggy ... Do you have any sample where you are doing resizing of
gif for tabItem... if so please share the sample project with me.

On Tue, May 23, 2023 at 12:09 PM DreamPiggy ***@***.***>
wrote:

> If the image class is SDAnimatedImage, you shouldn't use re-encode. The
> image data buffer is available from animatedImageData property.
>
> If you want to re-scale a SDAnimatedImage (I don't think people do this,
> but still possible). You need for-in loop to grap each frame and call that
> encodedData(with:frames) animated image encode API. See more in PR
> description of 5.15.0
>
> Or I can update this into the wiki
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SDWebImage/SDWebImage/discussions/3541#discussioncomment-5974510>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4AKH56HACRS3BPRIXWA3LXHRLRBANCNFSM6AAAAAAYG27VFQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Thanks And Regards
 Yogesh Gupta
